samples/sample_common/include/avc_headers.h

```diff
@@ -49,11 +49,8 @@ class HeaderSet
             m_header.resize(id + 1);
         }
 
-        if (m_header[id])
-        {
-            delete m_header[id];
-            m_header[id]=0;
-        }
+        delete m_header[id];
+        m_header[id]=0;
```

Comment:
you are assigning new value in next line. No need to zero it anyway (especially with ```int``` literal)